### DISCUSSION TITLE
Convergence issue at the transition between two blocks

### DISCUSSION THREAD

**Discussion by ZoeyChen1993** [idaholab/moose#21742] [Q&A Modules: General] [answered]
Dear experts,
I am setting up a simulation of a coupled thermal-chemical problem, the domain is a standard rectangle with two parts, the two parts are using different material properties, there is a heating source from the bottom, then the heat generated a reaction front, both heat and reaction front propagate along the blocks, then when they travel to the boundary between the two blocks, the mesh starts to adaptive and the time step goes to extremely small then solve does not converge.  Could some one help me to solve the convergence issue? The figure attached is a temperature contour when the simulation aborted. And the code from executioner to adaptivity blocks are attached below.
![test](https://user-images.githubusercontent.com/90401626/182254554-16736caf-785b-428d-bea3-05e6693d86fb.png)
```
[Executioner]
  type = Transient
  num_steps = 80000
  nl_rel_tol = 1e-6

  end_time=1000

  nl_max_its=10
  l_max_its=10
  
  [./TimeStepper]
    type = ConstantDT    
    dt = 0.5
    # dt = 3.6e14   
  [../]
  
  [./TimeIntegrator]
   type = ImplicitEuler
[../]
  
  #Preconditioned JFNK (default)
  solve_type = 'PJFNK'
  petsc_options_iname = '-pc_type -pc_hypre_type'
  petsc_options_value = 'hypre boomeramg'
[]


[Adaptivity]
  marker = combo #this line when commented, switches off adaptivity
  max_h_level = 3
  #steps = 2 #this line gets ignored in a transient run
  [./Indicators]
    [./Aerror]
      type = GradientJumpIndicator
      variable = Cure
      outputs = none
    [../]

    [./Terror]
      type = GradientJumpIndicator
      variable = Temperature
      outputs = none
    [../]
    [./error]
      type = GradientJumpIndicator
      variable = Temperature
      outputs = none
    [../]
    
  [../]
  [./Markers]
    [./Marker1]
      type = ErrorFractionMarker
      refine = 0.5
      coarsen = 0.1
      indicator = Aerror
      outputs = none
    [../]

    [./Marker3]
      type = ErrorFractionMarker
      refine = 0.5
      coarsen = 0.1
      indicator = Terror
      outputs = none
    [../]

    [./combo]
      type = ComboMarker
      markers = 'Marker1 Marker3'
    [../]
    
  [../]
  [./Markers]
    [./errorfrac]
      type = ErrorFractionMarker
      refine = 0.5
      coarsen = 0.1
      indicator = error
      outputs = none
    [../]
    
  [../]
[]
```

**Comment by GiudGiud** [comment, marked as answer]
Hello

It could be multiple things. Here are a few things I could see as issues

- The rel_tol of 1e-6 on non-linear can be very coarse for some problems. Typically you would want to look at adding automatic scaling to make sure (a little bit, you still need to verify yourself) you are converged for all variables in the simulation
- The vast differences in cell sizes can be an issue for some physics. Maybe you should limit the adaptivity more to see if you can get a solution. You may need to refine the time step to be able to handle very small cells (again, depends on the pjysics)
- The max number of iterations in your solve are small, so the solver does not get a lot of iterations to try to converge the problem before declaring a solve failure. You will want to raise both of these 10x to see if it can do better with more iterations
```
  nl_max_its=10
  l_max_its=10
```

Finally, if you have coded your Jacobian well, you should give the Newton method a try instead of PJFNK. It usually converges better

Hope this helps,
Guillaume
